falcon/errors.py

```diff
@@ -551,7 +583,12 @@ class HTTPMethodNotAllowed(HTTPError):
 
     @deprecated_args(allowed_positional=1)
     def __init__(
-        self, allowed_methods, title=None, description=None, headers=None, **kwargs
+        self,
+        allowed_methods: List[str],
```

Comment:
`iterable[str]` is probably ok, same as challenges above

Reply:
Changed.